src/sysen2/mldev.109

```diff
@@ -120,50 +121,24 @@ GO:	MOVE P,[-LPDLL-1,,PDL-1]
 		jrst gothst ]	; and go process request for this host
 
 nox:				; jump here if hostname minus first "x" is not a known ITS
-				; check to see if format is hhDIR
-	move c,a
-	lsh c,12.		; move past hostname
-	camn c,dirdev		; is what remains the DIR device?
-	  jrst dirchk		;  yes, continue checking hhDIR case
+	;; we know that we don't have the DIRhh, DIRhhh, nor hh case.  Check for hhh case.
+	hrlz c,a	
+	jumpe c,dohh		; we have HHH, but handling is the same as HH
 
-				; is it hhhdir or hhh
-	hrlz c,a		; get device in c
-	jumpe c, host3		; jump if no device
+	;; now we know we have either HHDDD, HHHDDD
+	;; check last character to see which case we have
 
 	ldb c,[000600,,a]	; can only be hhddd if last sixbit character is null
-	jumpe c,hhddd1		; handle hhddd case if last character is NUL
-	
-	hrlz c,a		; check for hhhddd, get ddd in c
-	jumpn e,hhhddd		; is there a ddd?, if so handle it
-	jrst chkits		; nope, assume that we just have hhh
-
-host3:				; seems like hhh, but may be special, check to see if ITS
-	move b,a		; hostname in A
-	pushj p,isitsp		; is it a known ITS
-	  jrst notits		;  nope, see if special device
-	move c,dskdev		; we have an ITS name, so assume DSK device
-	movem c,jbcdev		; save remote device
-	jrst gothst		; and process our host
-
-dirchk:				; 2-char hostname with DIR device
-	ldb b,[.bp (777700),a]	; get hh part in b
-	lsh b,24.		; left justify hostname
-dirch2:	move c,dirdev		; set remote device as DIR device
-	movem c,jbcdev
-	move a,b		; get host in a
-	jrst gothst		; process host
-
-hhddd2: ldb a,[301400,,jbcdev]	; get hostname in A
-	lsh a,24.		; left justify hostname
-	movem c,jbcdev		; set remote device from C
-	jrst gothst		; process host
+	jumpn c,hhhddd		; is there a ddd?, if so handle it
 
-hhddd1:				; continue check for hhddd
 	ldb c,[003000,,a]	; get device portion
 	lsh c,12.		; left-justify
-	jumpn c,hhddd2		; jump if we have a device; get hostname and set remote device
+	jumpe c,chkits		; if no device, check to see if HH is a host
```

Comment:
You already checked for empty device in HH case up on line 104.

Reply:
Ok, I removed the JUMPE C, CHKITS.  

Ah, I see why you mentioned BUBUNV 78, @bictorv.  I rebased my branch with origin/master and got this BUBUNV change that had already been integrated, I assume with the master branch.  Not sure why it shows up as a change here, since it should already be in master.  @larsbrinkhoff ?

In any case, I'll still wait until both @bictorv and @larsbrinkhoff approve this change before merging.  I've learned my lesson.  Also, I assume @larsbrinkhoff will want to combine some commits -- I've kept them separate to allow for review of the incremental changes made given the review comments.